_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,639 @@
+Which makes sense since the tests (unlike normal use), is being run by maintainers of the package itself, not its users.
+Though it still doesn't make a huge amount of sense to me, since spam from deprecation warnings floods out your actual errors that you want to see during testing.
+For this reason Invenia (my employer) has disabled deprecation warnings in the tests for all our closed source packages, and added a new test job that just does deprecation checking (set to error on deprecation and with the job allowed to fail, just so we have notice).
+
+Hopefully one day we can improve the tooling around deprecation warnings.
+An ideal behavour would be to only show deprecation warning if directly caused by a function call made from within the package module of your current active environment.
+I kind of know what we need to do to the logger to make that possible.
+But it is not yet something I have had time to do.
+
+
+### Code highlighting for `code_llvm` and `code_native`
+This was added in [#36634](https://github.com/JuliaLang/julia/issues/36634).
+This functionality was first implemented in the [ColoredLLCodes.jl](https://github.com/kimikage/ColoredLLCodes.jl), where it worked by monkey-patching the InteractiveUtils stdlib.
+That package does still work on julia 1.0.
+
+
+**Julia 1.0:**
+
+![Julia 1.0 code_llvm]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.0-code-llvm.png)
+
+**Julia 1.6:**
+
+![Julia 1.6 code_llvm]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.6-code-llvm.png)
+
+The REPL itself still doesn't have syntax highlighting for Julia code though.
+The [OhMyRepl.jl](https://github.com/KristofferC/OhMyREPL.jl) package does provide that, and works all versions of Julia.
+It is a lot more than a series of regex though, so I don't think we are going to see it built into julia too soon.
+Probably one day, though, as its big dependencies are also required if the parser wants to move to be written in julia.
+(Though I also don't expect that any time soon)
+
+### Clearer Stacktraces
+
+Just like colored `code_llvm` colored stack-traces also originated in a package that was doing some nasty monkey-patching: [ClearStacktrace.jl](https://github.com/jkrumbiegel/ClearStacktrace.jl).
+This was added into Base itself in [#36134](https://github.com/JuliaLang/julia/pull/36134)
+
+**Julia 1.0:**
+
+![Julia 1.0 stacktrace]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.0-stacktrace.png)
+
+**Julia 1.6:**
+
+![Julia 1.6 stacktrace]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.6-stacktrace.png)
+
+The first thing you probably notice is the colored package names, to make it more clear where the error is coming from.
+You also should notice the dimming of type-parameters to make complicated types easier to read.
+Also the addition of argument names, and showing keyword arguments bearing functions as they are written, rather than as a weird `#DataFrame#654` internal function.
```

Comment:
```suggestion
Also the addition of argument names, and showing functions with keyword arguments as they are written, rather than as a weird `#DataFrame#654` internal function.
```